src/core/device.c

```diff
@@ -766,6 +766,26 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (u) {
+                        if (UNIT_VTABLE(u)->active_state(u) == UNIT_ACTIVE) {
```

Comment:
These two if statements can be merged.

Reply:
I'll have a look, I kinda remember seperating them for a good reason, but i'll double check.